app/controllers/finance/payment_breakdowns_controller.rb

```diff
@@ -46,17 +46,19 @@ def choose_provider_npq
     end
 
     def choose_npq_statement
+      cohort = Cohort.find_by(start_year: params[:cohort_year])
       npq_lead_provider = NPQLeadProvider.find(params[:npq_lead_provider])
       statement_name = params[:statement].humanize.gsub("-", " ")
-      statement = npq_lead_provider.statements.find_by(name: statement_name)
+      statement = npq_lead_provider.statements.find_by(cohort:, name: statement_name)
 
       redirect_to finance_npq_lead_provider_statement_path(npq_lead_provider.id, statement)
     end
 
     def choose_ecf_statement
+      cohort = Cohort.find_by(start_year: params[:cohort_year])
       lead_provider = LeadProvider.find(params[:lead_provider])
       statement_name = params[:statement].humanize.gsub("-", " ")
-      statement = lead_provider.statements.find_by(name: statement_name)
+      statement = lead_provider.statements.find_by(cohort:, name: statement_name)
```

Comment:
whilst we don't want to implement any js to hide statements not available to lead providers, the current journey is a bit broken and returns a 500 if a statement is missing. Can we instead redirect and show a flash message saying statement not found if the statement is `nil` rather than getting a 500?

Reply:
we could do this but I feel we wouldn't get much value from doing this. if there will be some work to align to designs and add the JS this change would be made somewhat redundant

Reply:
I found it really hard to navigate/test and kept hitting an error locally, I wonder if others will hit the same on other test envs when deployed. The fix is pretty small with if/else and showing a flash message. We can implement if we get feedback regarding that I guess and for now can keep as is